### PR TITLE
Add amp-analytics proxy sample

### PR DESCRIPTION
**Pull request by andreban**
The sample shows how to use a the publisher
server to use the same clientId on analytics
for the publisher website and AMP pages

### CODE REVIEW

**Comment by andreban** [comment]
Need to properly parse the content from the _ga cookie:

 The format of the _ga cookie is GA<Version>.<Domain Code>[-<Path Code>].<ClientId>.
  - Version is 1 for now. It may change at a later point.
  - Domain Code is the level at which cookies are written. It varies from site to site but if you are writing cookies to nytimes.comthen the value will be 2. For blog.nytimes.com the value will be 3 and so on. If you do not set a domain to write cookies on in analytics.js, the top level domain where cookies can be written is used. So the value is probably 2.
  - Path code and the dash preceding it are optional and only set if the path on which cookies are written is different from "/".
  - Everything that follows after is client id for now.

File: controllers/index.js [hunk 1]
```diff
@@ -21,6 +21,8 @@ var router = require('express').Router();
 router.use('/amp-access', require('./amp-access'));
 router.use('/amp-user-notification', require('./amp-user-notification'));
 router.use('/amp-analytics', require('./amp-analytics'));
+router.use('/*/collect', require('./amp-analytics/proxy'));
```
Comment (sebastianbenz) [inline controllers/index.js]: why not use the amp-analytics prefix?

**Comment by sebastianbenz** [comment]
Nice example! Can you add a section to the ready explaining that it exists and what it does?

File: controllers/index.js (hunk 1 above)
Comment (andreban) [inline controllers/index.js]: I'm intercepting the same path pattern used by google-analytics, so the only thing I have to do later is change the host and cid

Comment (sebastianbenz) [inline controllers/index.js]: I see - makes sense

**Comment by andreban** [comment]
PTAL

Changed to use the UserId approach documented here: https://support.google.com/analytics/answer/3123662

Need to split the README.md into different sections (and files), explaining each sample. I'd rather do it on a different PR.

**Comment by sebastianbenz** [comment]
LGTM

**Pull request merged**